core/src/com/unciv/ui/tilegroups/WorldTileGroup.kt

```diff
@@ -25,14 +25,17 @@ class WorldTileGroup(internal val worldScreen: WorldScreen, tileInfo: TileInfo,
 
         icons.removePopulationIcon()
         val tileIsViewable = isViewable(viewingCiv)
+        val showEntireMap = UncivGame.Current.viewEntireMapForDebug
+
         if (tileIsViewable && tileInfo.isWorked() && UncivGame.Current.settings.showWorkedTiles
                 && city!!.civInfo == viewingCiv)
             icons.addPopulationIcon()
-
-        val currentPlayerCiv = worldScreen.viewingCiv
-        if (UncivGame.Current.viewEntireMapForDebug
-                || currentPlayerCiv.exploredTiles.contains(tileInfo.position))
-            updateCityButton(city, tileIsViewable || UncivGame.Current.viewEntireMapForDebug) // needs to be before the update so the units will be above the city button
+        // update city buttons in explored tiles or entire map
+        if (showEntireMap || viewingCiv.exploredTiles.contains(tileInfo.position))
+            updateCityButton(city, tileIsViewable || showEntireMap) // needs to be before the update so the units will be above the city button
+        else if (UncivGame.Current.gameInfo.currentPlayerCiv.isSpectator())
```

Comment:
I don't like the constant references to things trough the UncivGame.Current, that's to be avoided when possible.
I think this can be activated anyway, even if the current player isn't a spectator, no?

Reply:
> I don't like the constant references to things trough the UncivGame.Current, that's to be avoided when possible.

Sorry, always forget about this. Will get rid of this static link.

> I think this can be activated anyway, even if the current player isn't a spectator, no?

Well, I added this additional check just for performance issues, because in the normal mode, we generally don't touch unexplored tiles until next turn. Only in spectator mode where we can change views we can access them on each update.